### DISCUSSION TITLE
Why is `grammar.js` expected in the project root directory?

### DISCUSSION THREAD

**Discussion by buckynbrocko** [tree-sitter/tree-sitter#3376] [Q&A] [answered]
Seems strange that `grammar.js` wouldn't be located in `src/` and I wondered if maybe there was specific reasoning.

Mostly asking for my own curiosity.

**Comment by amaanq** [comment, marked as answer]
There's no major reason for this, but src is more for the source code artifacts that is actually used when compiling a parser to a shared library, in this case grammar.json, parser.c, and scanner.c, and having `grammar.js` be in the root dir makes it easier to access/view in general, I wouldn't consider it "source code" in the traditional sense that parser.c (and optionally scanner.c) are.
